### PR TITLE
Add documentation for signing artifacts

### LINKED ISSUE #1502: Document signing (do we sign with SHA256?)
### Is your feature request related to a problem? Please describe

The [documentation on signing](https://github.com/opensearch-project/opensearch-build#signing-artifacts) is very sparse. We have a non-open-source signer black box, but even simple questions like "do we sign with SHA256" cannot be easily answered.

### Describe the solution you'd like

Document what is being signed, when, and how, with what keys, algorithms, etc.

### Describe alternatives you've considered

_No response_

### Additional context

Coming from https://github.com/opensearch-project/opensearch-build/issues/1452#issuecomment-1015767175 and https://github.com/opensearch-project/opensearch-build/issues/27#issuecomment-858058677

### PR DESCRIPTION
**Pull request by gaiksaya**
Signed-off-by: Sayali Gaikawad <gaiksaya@amazon.com>

### Description
Adds documentation for signing artifacts

### Issues Resolved
resolves #1502 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,12 +125,38 @@ See [src/test_workflow](./src/test_workflow) for more information.
...
+Anything can be signed using PGP signing eg: tarball, any type of file, etc. A .sig file will be returned containing the signature. OpenSearch and OpenSearch dashboards distributions, components such as data prepper, etc as well as maven artifacts are signed using PGP signing. You can see how to verify the signatures by visiting the opensearch website (https://opensearch.org/verify-signatures.html).
+
+
+**Windows:**
+
+Windows signing can be used to sign windows executables such as .msi, .msp, .msm, .cab, .dll, .exe, .appx, .appxbundle, .msix, .msixbundle, .sys, .vxd, .ps1, .psm1, and any PE file that is supported by Signtool.exe (https://docs.microsoft.com/en-us/dotnet/framework/tools/signtool-exe). Various windows artifacts such as SQL OBDC, opensearch-cli, etc are signed using this method. 
+Windows code signing uses EV (Extended Validated) code signing certificates.
+
+
+**MacOS:**
```
Comment (zelinh) [inline README.md:145]: Our signing workflow doesn't support signing for MacOS at this time but the signer client does.

Comment (gaiksaya) [reply README.md:145]: I think @prudhvigodithi is working on adding it. In that case I'll remove everything related to macos for now. We can add back when we add macos signing functionality

Comment (gaiksaya) [reply README.md]: Done! cc: @bbarani

File: README.md [hunk 2]
```diff
@@ -125,12 +125,33 @@ See [src/test_workflow](./src/test_workflow) for more information.
...
+Windows code signing uses EV (Extended Validated) code signing certificates.
+
+|  Types of signing/Details   | Digest           | Cipher  | Key Size|
+| ------------- |:-------------| :-----| :-----|
+| PGP      | SHA1 | AES-128 | 2048 |
+| Windows      | SHA256      |    RSA | |
+| RPM | SHA512      |    RSA | 4096 |
+
+
+**Signing RPM artifacts:**
```
Comment (zelinh) [inline README.md:152]: Also for RPM artifacts signing, I think we only signing them via jenkins `signArtifacts` library, which is not related to our internal signer client. Should we also include it here? 
Or we could separate this from PGP or windows signing (which uses signer client) cause in line 128 you say "We use the client for all types of signing"
WDYT? cc. @peterzhuamazon

Comment (gaiksaya) [reply README.md:152]: Signing RPM artifacts is a different category I have added that it is yet to be integrated with signer client in the description.

File: README.md [hunk 3]
```diff
@@ -125,12 +125,33 @@ See [src/test_workflow](./src/test_workflow) for more information.
 
 #### Signing Artifacts
 
-The signing step takes the manifest file created from the build step and signs all its component artifacts using a tool called `opensearch-signer-client` (in progress of being open-sourced). The input requires a path to the build manifest and is expected to be inside the artifacts directory with the same directories mentioned in the build step. 
+For all types of Signing within OpenSearch project we use `opensearch-signer-client` (in progress of being open-sourced) which is a wrapper around internal signing system and is only available for authenticated users. The input requires a path to the build manifest or directory containing all the artifacts or a single artifact. The tool currently supports following platforms for signing.
```
Comment (dblock) [inline README.md:128]: lowercase signing

File: README.md [hunk 4]
```diff
@@ -125,12 +125,33 @@ See [src/test_workflow](./src/test_workflow) for more information.
...
 
+Usage:
```
Comment (dblock) [inline README.md:130]: It says above ".. supports the following platforms for signing" but then has usage. Split the paragraph above into usage and platforms.

File: README.md [hunk 5]
```diff
@@ -125,12 +125,33 @@ See [src/test_workflow](./src/test_workflow) for more information.
...
 ```bash
 ./sign.sh builds/opensearch/manifest.yml
 ```
 
+**PGP**:
```
Comment (dblock) [inline README.md:135]: Make these ##### sections, so they appear in TOC

File: README.md [hunk 6]
```diff
@@ -125,12 +125,33 @@ See [src/test_workflow](./src/test_workflow) for more information.
...
+
+Anything can be signed using PGP signing eg: tarball, any type of file, etc. A .sig file will be returned containing the signature. OpenSearch and OpenSearch dashboards distributions, components such as data prepper, etc as well as maven artifacts are signed using PGP signing. You can see how to verify the signatures by visiting the opensearch website (https://opensearch.org/verify-signatures.html).
```
Comment (dblock) [inline README.md:137]: Link opensearch website - I would say `See [this page](https://opensearch.org/verify-signatures.html) for how to verify signatures.`

File: README.md [hunk 7]
```diff
@@ -125,12 +125,33 @@ See [src/test_workflow](./src/test_workflow) for more information.
...
+
+
+**Windows:**
+
+Windows signing can be used to sign windows executables such as .msi, .msp, .msm, .cab, .dll, .exe, .appx, .appxbundle, .msix, .msixbundle, .sys, .vxd, .ps1, .psm1, and any PE file that is supported by Signtool.exe (https://docs.microsoft.com/en-us/dotnet/framework/tools/signtool-exe). Various windows artifacts such as SQL OBDC, opensearch-cli, etc are signed using this method.
```
Comment (dblock) [inline README.md:142]: Use markdown link for the docs.

File: README.md [hunk 8]
```diff
@@ -125,12 +125,33 @@ See [src/test_workflow](./src/test_workflow) for more information.
...
+
+|  Types of signing/Details   | Digest           | Cipher  | Key Size|
+| ------------- |:-------------| :-----| :-----|
+| PGP      | SHA1 | AES-128 | 2048 |
+| Windows      | SHA256      |    RSA | |
+| RPM | SHA512      |    RSA | 4096 |
+
+
+**Signing RPM artifacts:**
+We have an issue (https://github.com/opensearch-project/opensearch-build/issues/1547) open to add RPM signing functionality to the above signing system. However, temporarily we are signing the RPM artifacts via shell script which uses a macros template (https://github.com/opensearch-project/opensearch-build/blob/main/scripts/pkg/sign_templates/rpmmacros). More details in this commit (https://github.com/opensearch-project/opensearch-build/commit/950d55c1ed3f82e98120541fa40ff506338c1059). Currently we are only signing OpenSearch and OpenSearch dashboards RPM distribution using this method.
```
Comment (dblock) [inline README.md:153]: Generally avoid using "we". 

You can say "RPMs are signed ..." and "See issue X ... for

**Review by dblock — COMMENTED** [review]
Some nits.

**Comment by gaiksaya** [comment]
@dblock Addressed all the comment. Please re-review. Thanks!

File: README.md [hunk 9]
```diff
@@ -125,12 +125,34 @@ See [src/test_workflow](./src/test_workflow) for more information.
...
+Windows code signing uses EV (Extended Validated) code signing certificates.
+
+|  Types of signing/Details   | Digest           | Cipher  | Key Size|
+| ------------- |:-------------| :-----| :-----|
+| PGP      | SHA1 | AES-128 | 2048 |
+| Windows      | SHA256      |    RSA | |
+| RPM | SHA512      |    RSA | 4096 |
+
+
+### Signing RPM artifacts:
```
Comment (dblock) [inline README.md:153]: Super nit: these titles don’t need : and Should be Capitalized

**Review by dblock — APPROVED** [review]

**Pull request merged**